fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/scalar/Substring.java

```diff
@@ -68,7 +68,7 @@ public Substring(Expression arg0, Expression arg1, Expression arg2) {
     @Override
     public FunctionSignature computeSignature(FunctionSignature signature) {
         Optional<Expression> length = arity() == 3
-                ? Optional.of(getArgument(2))
+                ? Optional.of(getArgument(1))
```

Comment:
according [substr's manual](https://doris.apache.org/docs/dev/sql-manual/sql-functions/string-functions/substring/), the third arg is length. so we need to get length of return type from `getArgument(2)` or `getArgument(0).getDataType().len() - getArgument(1)`